test/e2e/tools/event-writer/event_writer.cpp

```diff
@@ -0,0 +1,329 @@
+        goto fail;
+    }
+    if (_pin(FILTER_MAP_PIN_PATH, bpf_map__fd(map_flt), true) != 0) {
+        goto fail;
+    }
+
+    printf("%s - event-writer loaded successfully\n", __FUNCTION__);
+    bpf_object__close(obj);
+    return 0; // Return success
+
+fail:
+    if (prg != NULL) {
+        bpf_program__unpin(prg, EVENT_WRITER_PIN_PATH);
+    }
+
+    if (map_ev != NULL) {
+        bpf_map__unpin(map_ev, EVENTS_MAP_PIN_PATH);
+    }
+
+    if (map_flt != NULL) {
+        bpf_map__unpin(map_flt, FILTER_MAP_PIN_PATH);
+    }
+
+    if (map_fvt != NULL) {
+        bpf_map__unpin(map_fvt, FIVE_TUPLE_MAP_PIN_PATH);
+    }
+
+    if (map_met != NULL) {
+        bpf_map__unpin(map_met, METRICS_MAP_PIN_PATH);
+    }
+
+    if (obj != NULL) {
+        bpf_object__close(obj);
+    }
+    return 1;
+}
+
+int
+unpin(void) {
+    if (bpf_obj_get(EVENT_WRITER_PIN_PATH) < 0) {
+        fprintf(stderr, "%s - failed to lookup event_writer at %s\n", __FUNCTION__, EVENT_WRITER_PIN_PATH);
+    } else {
+        ebpf_object_unpin(EVENT_WRITER_PIN_PATH);
+    }
+
+    if (bpf_obj_get(FILTER_MAP_PIN_PATH) < 0) {
+        fprintf(stderr, "%s - failed to lookup filter_map at %s\n", __FUNCTION__, FILTER_MAP_PIN_PATH);
+   } else {
+       ebpf_object_unpin(FILTER_MAP_PIN_PATH);
+   }
+
+    if (bpf_obj_get(EVENTS_MAP_PIN_PATH) < 0) {
+         fprintf(stderr, "%s - failed to lookup cilium_events at %s\n", __FUNCTION__, EVENTS_MAP_PIN_PATH);
+    } else {
+        ebpf_object_unpin(EVENTS_MAP_PIN_PATH);
+    }
+
+    if (bpf_obj_get(METRICS_MAP_PIN_PATH) < 0) {
+        fprintf(stderr, "%s - failed to lookup cilium_metrics at %s\n", __FUNCTION__, METRICS_MAP_PIN_PATH);
+    } else {
+        ebpf_object_unpin(METRICS_MAP_PIN_PATH);
+    }
+
+    if (bpf_obj_get(FIVE_TUPLE_MAP_PIN_PATH) < 0) {
+        fprintf(stderr, "%s - failed to lookup five_tuple_map at %s\n", __FUNCTION__, FIVE_TUPLE_MAP_PIN_PATH);
+    } else {
+        ebpf_object_unpin(FIVE_TUPLE_MAP_PIN_PATH);
+    }
+
+    return 0;
+ }
+
+uint32_t _ipStrToUint(const char* ipStr) {
+    uint32_t ip = 0;
+    int part = 0;
+    int parts = 0;
+    const char *p = ipStr;
+    char c;
+
+    while ((c = *p++) != '\0') {
+        if (c >= '0' && c <= '9') {
+            part = part * 10 + (c - '0');
+        } else if (c == '.') {
+            ip = (ip << 8) | (part & 0xFF);
+            part = 0;
+            parts++;
+        } else {
+            // Invalid character in IP string.
+            return 0;
+        }
+    }
+
+    // Process the last octet.
+    ip = (ip << 8) | (part & 0xFF);
+    parts++;
+
+    // Ensure we have exactly four parts
+    if (parts != 4) {
+        return 0;
+    }
+
+    return ip;
+}
+
+int main(int argc, char* argv[]) {
+    setvbuf(stdout, NULL, _IONBF, 0);
+    // Parse the command-line arguments (flags)
+    if (argc < 2) {
+        fprintf(stderr, "valid arguments are required. Exiting..\n");
```

Comment:
suggest: document what combination of arguments are accepted. We should at least have a comment that is doing this, but also having it as an output in case of error is good.

Consider all below cases of invalid arguments to also print this error message.